src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HealthHeartbeatProvider.cs

```diff
@@ -0,0 +1,226 @@
+        }
+
+        public void UnregisterHeartbeatPayload(string providerName)
+        {
+            if (this.payloadItems != null)
+            {
+                this.payloadItems.Remove(providerName);
+            }
+        }
+
+        #region IDisposable Support
+
+        // Override the finalizer only if Dispose(bool disposing) above has code to free unmanaged resources.
+        // ~HealthHeartbeatProvider() {
+        //   // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
+        //   Dispose(false);
+        // }
+
+        // This code added to correctly implement the disposable pattern.
+        public void Dispose()
+        {
+            // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
+            this.Dispose(true);
+            // uncomment the following line if the finalizer is overridden above.
+            // GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!this.disposedValue)
+            {
+                if (disposing)
+                {
+                }
+
+                // free unmanaged resources (unmanaged objects) and override a finalizer below.
+                // set large fields to null.
+
+                this.disposedValue = true;
+            }
+        }
+
+        #endregion
+
+        protected void Send()
+        {
+            this.SendHealthHeartbeat();
+        }
+
+        protected MetricTelemetry GatherData()
+        {
+            return this.GatherDataForHeartbeatPayload();
+        }
+
+        protected void AddDefaultPayloadItems(IDictionary<string, IHealthHeartbeatPayloadExtension> heartbeatPayloadItems)
+        {
+            try
+            {
+                heartbeatPayloadItems[string.Empty] = new HealthHeartbeatDefaultPayload(this.enabledHeartbeatPayloadFields);
+            }
+            catch (Exception e)
+            {
+                throw new ArgumentException("HealthHeartbeatProvider::AddDefaultPayloadItems : Unable to add default payload items to health heartbeat", nameof(heartbeatPayloadItems), e);
+            }
+        }
+
+        private MetricTelemetry GatherDataForHeartbeatPayload()
+        {
+            var heartbeat = new MetricTelemetry(HeartbeatSyntheticMetricName, 0.0);
+
+            foreach (var payloadItem in this.payloadItems)
+            {
+                var props = payloadItem.Value.GetPayloadProperties();
+                foreach (var kvp in props)
+                {
+                    heartbeat.Properties.Add(kvp.Key, kvp.Value.ToString());
+                }
+
+                heartbeat.Sum += payloadItem.Value.CurrentUnhealthyCount;
+                heartbeat.Sequence = string.Format(CultureInfo.InvariantCulture, "{0}", this.heartbeatsSent++);
+            }
+
+            return heartbeat;
+        }
+
+        private void SendHealthHeartbeat()
+        {
+            try
```

Comment:
We need to send heartbeat even if almost no payload were collected. Otherwise bad provider will not allow to send any heartbeat details and we will lack any information about SDK..

So I suggest to create a telemetry object in this method and give it to providers so they'd add details

Reply:
Good idea.   Will incorporate that into the updated design/implementation.